### ISSUE TITLE
FMA on Windows is weirdly broken

### ISSUE THREAD

**Issue by oscardssmith** [JuliaLang/julia#43088]
```
julia> func=fma;
julia> @test func(-1.9369631f13, 2.1513551f-7, -1.7354427f-24) == -4.1670958f6
Test Passed

julia> for func in (fma, fma)
           @test func(-1.9369631f13, 2.1513551f-7, -1.7354427f-24) == -4.1670958f6
       end
Test Failed at REPL[37]:2
  Expression: func(-1.9369631f13, 2.1513551f-7, -1.7354427f-24) == -4.1670958f6 
   Evaluated: -4.1670955f6 == -4.1670958f6
ERROR: There was an error during testing 
```
I'm really not sure how this is possible (This is the case both on 1.6 and master)

**Comment by KristofferC** [comment]
Is this https://github.com/JuliaLang/julia/issues/41450?

**Comment by oscardssmith** [comment]
Itś related but not quite the same. That issue only affected `muladd`, where you can make an argument that the observed behavior is correct (although it´s still bad). Here the bug happens for `fma` where it 100% a bug.

**Comment by oscardssmith** [comment]
https://github.com/MicrosoftDocs/cpp-docs/pull/3526/files

**Comment by vtjnash** [comment]
That one doesn't affect us currently. This is a bug in the mingw-w64 compiler libraries (https://sourceforge.net/p/mingw-w64/bugs/848/)

**Comment by JeffreySarnoff** [comment]
there is a response in the mingw-w64 bug note [from Jameson]
(https://sourceforge.net/p/mingw-w64/bugs/848/#be9f) that suggests two alternative implementations -- should we use one of them meanwhile?

**Comment by oscardssmith** [comment]
we do. The problem is that LLVM is constant propagating things to the wrong answer and we can't do anything about that.

**Comment by JeffreySarnoff** [comment]
Is there a way to use fma now that works correctly on Windows?

**Comment by JeffreySarnoff** [comment]
I am concerned that DoubleFloats, which relies on fma everywhere, could be giving incorrect results when used on Windows!

**Comment by oscardssmith** [comment]
TLDR is that windows is only ever off by 1 ULP (as far as I can tell), so DoubleFloats will still give mostly right answers. Also, the error compensation uses of FMA have some relatively nice properties that mean that fma might always produce the right answer for those fmas (this is purely conjecture though. I haven't done sufficient analysis to confirm this).

**Comment by vtjnash** [comment]
Isn't this also only an issue if we remove openlibm?

**Comment by oscardssmith** [comment]
no. It's an issue now. It's been an issue since at least 1.6 (and probably way earlier).

**Comment by vtjnash** [comment]
IIUC, this is addressed now
